### PR TITLE
add support for kotlin multiplatform

### LINKED ISSUE #41: Support multiplatform
We can migrate to K2 and support Compose Multiplatform, it would be great to publish this lib to MavenCentral.

### PR DESCRIPTION
**Pull request by Rafael2616**
this commit adds support for most kotlin multiplatform targets including ios, wasm, js, jvm also fixes https://github.com/nanihadesuka/LazyColumnScrollbar/issues/41

all tests have passed
compiled for all platform and got successful

* setup targets
* move tests from androidTest to androidUnitTest sourceSet, some dependencies are android-specific and we cant put it on commonTest sourceSet
* moved previews to androidMain sourceSet as it is android-specific
* migrated some components to material3
* bumped Kotlin to 2.0.20
* bumped agp to 8.6.0
* remove explicit compose buildFeature declaration as is is no more needed after AGP 8.5.2
* bumped some dependencies to their latest version
* get rid on some unused dependencies like appcompat etc..
* use jetbrains compose compiler and jetbrains compose dependencies

### CODE REVIEW

**Comment by Rafael2616** [comment]
Also the ":app" module ramains Android only. didn't think it was a good idea to migrate it at this time, maybe in another pull

**Comment by nanihadesuka** [comment]
Wow 🌞 

Sorry for the lateness. Looks very promising! Will take a look, and also will look at KMP topics before as I don't have much experience at it to make a proper review. Give me 1 week to look at the PR

**Comment by Rafael2616** [comment]
Sure, no problem, If you have any questions just ask, I'm already using this in my app, but I had to embed it in this one, so I decided to contribute 😅

File: kotlin-js-store/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,2820 @@
...
+  version "1.2.1"
+  resolved "https://registry.yarnpkg.com/@jridgewell/set-array/-/set-array-1.2.1.tgz#558fb6472ed16a4c850b889530e6b36438c49280"
+  integrity sha512-R8gLRTZeyp03ymzP/6Lil/28tGeGEzhx1q2k703KGWRAI1VdvPIXdG70VJc2pAMw3NA6JKL5hhFu1sJX0Mnn/A==
+
+"@jridgewell/source-map@^0.3.3":
+  version "0.3.6"
+  resolved "https://registry.yarnpkg.com/@jridgewell/source-map/-/source-map-0.3.6.tgz#9d71ca886e32502eb9362c9a74a46787c36df81a"
+  integrity sha512-1ZJTZebgqllO79ue2bm3rIGud/bOe0pP5BjSRCRxxYkEZS8STV7zN84UBbiYu7jy+eCKSnVIUgoWWE/tt+shMQ==
+  dependencies:
+    "@jridgewell/gen-mapping" "^0.3.5"
```
Comment (nanihadesuka) [inline kotlin-js-store/yarn.lock:39]: This is listed as malware by some users
https://github.com/github/advisory-database/issues/3280
Know from where this comes from?

File: kotlin-js-store/yarn.lock
Comment (nanihadesuka) [inline kotlin-js-store/yarn.lock:1]: I suppose this is autogenerated

File: lib/build.gradle.kts [hunk 2]
```diff
@@ -71,16 +71,66 @@ afterEvaluate {
...
-    implementation(libs.androidx.compose.material)
-    implementation(libs.androidx.compose.ui.ui)
-    implementation(libs.androidx.compose.ui.tooling.preview)
+kotlin {
 
-    debugImplementation(libs.androidx.compose.ui.test.manifest)
-    debugImplementation(libs.androidx.compose.ui.tooling)
+    val isAndroidApp = plugins.hasPlugin("com.android.application")
+    val isAndroidLibrary = plugins.hasPlugin("com.android.library")
+    if (isAndroidApp || isAndroidLibrary) {
```
Comment (nanihadesuka) [inline lib/build.gradle.kts]: The lib module will never be a android app module. Is isAndroidApp necessary?

File: kotlin-js-store/yarn.lock (hunk 1 above)
Comment (Rafael2616) [reply kotlin-js-store/yarn.lock:39]: this is an autogenerated file by kotlin, an the versions listed there cant be directly updated/removed. In each new Kotlin update just run :kotlinUpgradeYarnLock task to get new dependency updates

Comment (Rafael2616) [reply kotlin-js-store/yarn.lock:39]: im not sure if this an big problem

File: kotlin-js-store/yarn.lock
Comment (Rafael2616) [reply kotlin-js-store/yarn.lock:1]: yes

File: lib/build.gradle.kts (hunk 2 above)
Comment (Rafael2616) [reply lib/build.gradle.kts]: you´re correct. it can safely removed!

**Comment by nanihadesuka** [comment]
Thanks for the work. I will merge then do a cursory check, update readme & publish it the following weekend

**Review by nanihadesuka — APPROVED** [review]

**Pull request merged**